Example/Tests/BONUtilitiesTestCase.m

```diff
@@ -0,0 +1,252 @@
+        NSFontAttributeName : [UIFont systemFontOfSize:16],
+    }];
+
+    BONAssertAttributedStringHasAttributes(textField.attributedText, controlAttributes);
+}
+
+- (void)testTextFieldTextBeforeChainable
+{
+    UITextField *textField = UITextField.new;
+    textField.textAndApplyChainable = @"Hello, world!";
+    textField.bonChainable = BONChain.new.font([UIFont systemFontOfSize:16]);
+
+    XCTAssertEqualObjects(textField.attributedText.string, @"Hello, world!");
+
+    NSMutableDictionary *controlAttributes = [self defaultAttributesForClass:[UITextField class] withString:@"Hello, world!"];
+    [controlAttributes[BONValueFromRange(0, 13)] addEntriesFromDictionary:@{
+        NSFontAttributeName : [UIFont systemFontOfSize:16],
+    }];
+
+    BONAssertAttributedStringHasAttributes(textField.attributedText, controlAttributes);
+}
+
+- (void)testTextFieldAttributedTextAfterChainable
+{
+    UITextField *textField = UITextField.new;
+    textField.bonChainable = BONChain.new.font([UIFont systemFontOfSize:16]);
+    textField.attributedText = [[NSAttributedString alloc] initWithString:@"Hello, world!"];
+
+    XCTAssertEqualObjects(textField.attributedText.string, @"Hello, world!");
+
+    NSDictionary *controlAttributes = [self defaultAttributesForClass:[UITextField class] withString:@"Hello, world!"];
+
+    BONAssertAttributedStringHasAttributes(textField.attributedText, controlAttributes);
+}
+
+- (void)testTextFieldAttributedTextBeforeChainable
+{
+    UITextField *textField = UITextField.new;
+    textField.attributedText = [[NSAttributedString alloc] initWithString:@"Hello, world!"];
+    textField.bonChainable = BONChain.new.font([UIFont systemFontOfSize:16]);
+
+    XCTAssertNotNil(textField.bonChainable);
+    XCTAssertEqualObjects(textField.attributedText.string, @"Hello, world!");
+
+    NSMutableDictionary *controlAttributes = [self defaultAttributesForClass:[UITextField class] withString:@"Hello, world!"];
+    [controlAttributes[BONValueFromRange(0, 13)] addEntriesFromDictionary:@{
+        NSFontAttributeName : [UIFont systemFontOfSize:16],
+    }];
+
+    BONAssertAttributedStringHasAttributes(textField.attributedText, controlAttributes);
+}
+
+#pragma mark - Helpers
+
+- (NSMutableDictionary *)defaultAttributesForClass:(Class)viewClass withString:(NSString *)string
+{
+    NSMutableDictionary *attributes = [NSMutableDictionary dictionary];
+
+    id view = [[viewClass alloc] init];
+    if ([view respondsToSelector:@selector(setAttributedText:)] &&
+        [view respondsToSelector:@selector(attributedText)]) {
+        [view setAttributedText:[[NSAttributedString alloc] initWithString:string]];
+        [[view attributedText] enumerateAttributesInRange:NSMakeRange(0, string.length) options:0 usingBlock:^(NSDictionary<NSString *, id> *_Nonnull attrs, NSRange range, BOOL *_Nonnull stop) {
+            attributes[[NSValue valueWithRange:range]] = [attrs mutableCopy];
+        }];
+    }
```

Comment:
#nth given that this is a test, we should probably assert in an `else` clause here, since that would reflect misuse of this utility method.